### PR TITLE
Feature/logging

### PR DESCRIPTION
**Pull request by dbmeneses**
Pretty small commits, so doing a single P/R

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/issue/tracking/IssueTransition.java [hunk 1]
```diff
@@ -66,9 +67,23 @@ public void execute() {
...
-    for (BatchComponent component : componentCache.all()) {
-      trackIssues(reader, component);
+    ProgressReport progressReport = new ProgressReport("issue-tracking-report", TimeUnit.SECONDS.toMillis(10));
+    progressReport.start("Performing issue tracking");
+    int count = 0;
+
+    try {
+      for (BatchComponent component : componentCache.all()) {
+        trackIssues(reader, component);
+        progressReport.message(++count + "/" + nbComponents + " components tracked");
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/tracking/IssueTransition.java]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)

File: sonar-batch/src/main/java/org/sonar/batch/bootstrapper/LoggingConfiguration.java [hunk 2]
```diff
@@ -74,16 +84,29 @@ public LoggingConfiguration setVerbose(boolean verbose) {
...
+    String deprecatedProfilingLevel = getFallback("sonar.log.profilingLevel", props, fallback);
+    boolean verbose = "true".equals(getFallback("sonar.verbose", props, fallback)) ||
       "DEBUG".equals(logLevel) || "TRACE".equals(logLevel) ||
       "BASIC".equals(deprecatedProfilingLevel) || "FULL".equals(deprecatedProfilingLevel);
 
     return setVerbose(verbose);
   }
 
+  @CheckForNull
+  private String getFallback(String key, Map<String, String> properties, @Nullable Map<String, String> fallback) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrapper/LoggingConfiguration.java]: MAJOR Make "getFallback" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by henryju** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request closed without merging**